scenarios/authentication/rest.go

```diff
@@ -0,0 +1,125 @@
+package rest
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/go-resty/resty/v2"
+)
+
+const (
+	kuberahost = "KUBERA_HOST"
+	kuberauser = "KUBERA_USER"
+	kuberapass = "KUBERA_PASS"
+)
+
+type KuberaToken struct {
+	JWT       string `json:"jwt"`
+	AccountID string `json:"accountId"`
+}
+
+type KuberaError struct {
+	Message string `json:"message"`
+}
+
+// TokenFetching contains the details of credentials
+type TokenFetching struct {
+	UserName string
+	Password string
+	Host     string
+}
+
+//RequestAccess have a credentials to access Kubera api
+type RequestAccess struct {
+	Token     string
+	AccountID string
+}
+
+// Response for defining the response of postmethod
+type Response struct {
+	Resp       *resty.Response
+	Token      string
+	Message    string
+	Error      error
+	StatusCode int
+	AccountID  string
+}
+
+//NewTokensier return the credentials of kubera user ,pass and host
+func NewTokensier() (*TokenFetching, error) {
+	host, ok := os.LookupEnv(kuberahost)
+	if !ok {
+		return nil, errors.New("KHOST environment variable required")
+	}
+	userName, ok := os.LookupEnv(kuberauser)
+	if !ok {
+		return nil, errors.New("KUSER environment variable required")
+	}
+	userPass, ok := os.LookupEnv(kuberapass)
+	if !ok {
+		return nil, errors.New("KPASS environment variable required")
+	}
+	fmt.Println(host, userName, userPass)
+	return &TokenFetching{
+		Host:     host,
+		UserName: userName,
+		Password: userPass,
+	}, nil
+}
+
+func (a *TokenFetching) FetchKuberaToken() (*Response, error) {
+	client := resty.New()
+	url := a.Host + "v3/token"
+	token := KuberaToken{}
+	kErr := KuberaError{}
+	bodyData := map[string]interface{}{
+		"code":         fmt.Sprintf("%s:%s", a.UserName, a.Password),
+		"authProvider": "localAuthConfig",
+	}
+	resp, err := client.R().
+		SetHeader("Content-Type", "application/json").
+		SetBody(bodyData).
+		SetError(&kErr).
```

Comment:
```go
SetError(&errmsg)
```

Reply:
What is the argument's datatype required by `SetError` method?

Reply:
Understand the difference between SetError & err response

Reply:
_**SetError**_ method is to register the global or client common `Error` object into Resty. It is used for automatic unmarshalling if response status code is greater than 399 and content type either JSON or XML. Can be pointer or non-pointer.

_**SetResult**_ method is to register the response `Result` object for automatic unmarshalling for the request, if response status code is between 200 and 299 and content type either JSON or XML.